packages/aws-cdk-lib/aws-route53/lib/record-set.ts

```diff
@@ -929,6 +929,10 @@ export class CrossAccountZoneDelegationRecord extends Construct {
       throw Error('Only one of parentHostedZoneName and parentHostedZoneId is supported');
     }
 
+    if (!props.delegatedZone.hostedZoneNameServers) {
+      throw Error(`Not able to retrieve Name Servers for ${props.delegatedZone.zoneName} due to it being imported.`);
```

Comment:
will it always be imported for this construct?

Reply:
Are you asking about the "due to it being imported" part of the error message? And should we assume `delegatedZone` is imported?

If so, yes, afaik only imported Hosted Zone would have `hostedZoneNameServers` being undefined.

Reply:
thats what i was wondering, thanks